stdlib/string.mc

```diff
@@ -29,6 +31,63 @@ let int2string = fix (lam int2string. lam n.
     concat (int2string (divi n 10)) d
 )
 
+// Returns an option with the index of the first occurrence of c in s. Returns
+// None if c was not found in s.
+let strindex = lam c. lam s.
+  let strindex_abs = fix (lam strindex. lam i. lam c. lam s.
```

Comment:
Minor comment: the `foo_abs` naming convention is something I used in the experiments where I was separating `fix` from definition. It does not "mean" anything in particular.